document/js-api/index.bs

```diff
@@ -687,6 +695,90 @@ which can be simultaneously referenced by multiple {{Instance}} objects. Each
+        1. Return |map|[|globaladdr|].
+    1. Let |global| be a new {{Global}} instance with \[[Global]] set to |globaladdr|.
+    1. [=map/Set=] |map|[|globaladdr|] to |global|.
+    1. Return |global|.
+</div>
+
+<div algorithm>
+    To <dfn>ToValueType</dfn>(|s|), perform the following steps:
+    1. If |s| equals "i32", return [=𝗂𝟥𝟤=].
+    1. If |s| equals "f32", return [=𝖿𝟥𝟤=].
+    1. If |s| equals "f64", return [=𝖿𝟨𝟦=].
+    1. Otherwise, throw a {{TypeError}} exception.
+</div>
+
+<div algorithm>
+    The <dfn constructor for="Global">Global(descriptor)</dfn> constructor, when invoked, performs the following steps:
+    1. If |descriptor|["value"] is [=present=], let |v| be |descriptor|["value"]; otherwise, let |v| be 0.
+    1. If |descriptor|["mutable"] is [=present=], let |mutable| be |descriptor|["mutable"]; otherwise, let |mutable| be false.
+    1. Let |type| be [=ToValueType=](|descriptor|["type"]).
+    1. Let |value| be [=ToWebAssemblyValue=](|v|, |type|).
+    1. If |mutable| is true, let |globaltype| be [=var=] |type|; otherwise, let |globaltype| be [=const=] |type|.
+    1. Let |store| be the current agent's [=associated store=].
+    1. Let (|store|, |globaladdr|) be [=alloc_global=](|store|, |globaltype|, |value|). <!-- TODO(littledan): Report allocation failure https://github.com/WebAssembly/spec/issues/584 -->
+    1. Set the current agent's [=associated store=] to |store|.
+    1. [=Create a global object=] from the global address |globaladdr| and return the result.
+</div>
+
+<div algorithm>
+    The getter of the <dfn attribute for="Global">value</dfn> attribute of {{Global}}, when invoked, performs the following steps:
+    1. Let |global| be the Global instance.
+    1. Let |store| be the current agent's [=associated store=].
+    1. Let |globaladdr| be |global|.\[[Global]].
+    1. Let |globaltype| be [=type_global=](|store|, |globaladdr|).
+    1. If |type| is [=𝗂𝟨𝟦=], throw a {{TypeError}}.
+    1. Let |value| be [=read_global=](|store|, |globaladdr|).
+    1. Return [=ToJSValue=](|value|).
+
+    The setter of the value attribute of {{Global}}, when invoked with a value |v|, performs the following steps:
+    1. Let |global| be the Global instance.
+    1. Let |store| be the current agent's [=associated store=].
+    1. Let |globaladdr| be |global|.\[[Global]].
+    1. Let |globaltype| be [=type_global=](|store|, |globaladdr|).
+    1. If |globaltype| is of the form [=const=] |type|, throw a {{TypeError}.
+    1. If |type| is [=𝗂𝟨𝟦=], throw a {{TypeError}}.
+    1. Let |value| be [=ToWebAssemblyValue=](|v|, |type|).
+    1. Let |store| be [=write_global=](|store|, |globaladdr|, |value|).
+    1. If |store| is [=error=], throw a {{RangeError}} exception.
+    1. Set the current agent's [=associated store=] to |store|.
+</div>
+
+<div algorithm>
+    The <dfn method for="Global">valueOf()</dfn> method, when invoked, returns the result of invoking the getter for {{Global/value}}.
```

Comment:
This is not generally great practice; you should ideally define a shared algorithm that takes a Global instance and call it from both places. Right now, as written, one could imagine this changing in response to `Object.defineProperty(Global.prototype, "value", { get() { ... arbitrary code here ... })`.